packages/flutter/lib/src/widgets/routes.dart

```diff
@@ -1832,14 +1832,6 @@ abstract class ModalRoute<T> extends TransitionRoute<T> with LocalHistoryRoute<T
     ];
   }
 
-  @override
```

Comment:
looking at the code, it looks like popscope/willpopscope are going through Navigator.maybePop mechanism and are still allowed to be popped through Navigator.pop. So it is not handle pop internally.